Build issue on macOS 10.13.2 - develop branch

Dear All,

I am trying to build the OpenBLAS development branch on a Mac system, but there appears to be an extraneous (or incomplete) archive call when compiling the Lapack routines:

```bash
ar -ru ../../../libopenblas_haswellp-r0.3.0.dev.a lapacke_cggsvp.o lapacke_cggsvp_work.o lapacke_dggsvp.o lapacke_dggsvp_work.o lapacke_sggsvp.o lapacke_sggsvp_work.o lapacke_zggsvp.o lapacke_zggsvp_work.o lapacke_cggsvd.o lapacke_cggsvd_work.o lapacke_dggsvd.o lapacke_dggsvd_work.o lapacke_sggsvd.o lapacke_sggsvd_work.o lapacke_zggsvd.o lapacke_zggsvd_work.o lapacke_cgeqpf.o lapacke_cgeqpf_work.o lapacke_dgeqpf.o lapacke_dgeqpf_work.o lapacke_sgeqpf.o lapacke_sgeqpf_work.o lapacke_zgeqpf.o lapacke_zgeqpf_work.o
ar -ru ../../../libopenblas_haswellp-r0.3.0.dev.a 
ar: no archive members specified
usage:  ar -d [-TLsv] archive file ...
	ar -m [-TLsv] archive file ...
	ar -m [-abiTLsv] position archive file ...
	ar -p [-TLsv] archive [file ...]
	ar -q [-cTLsv] archive file ...
	ar -r [-cuTLsv] archive file ...
	ar -r [-abciuTLsv] position archive file ...
	ar -t [-TLsv] archive [file ...]
	ar -x [-ouTLsv] archive [file ...]
make[3]: *** [../../../libopenblas_haswellp-r0.3.0.dev.a] Error 1
make[2]: *** [lapacke] Error 2
make[1]: *** [lapackelib] Error 2
make: *** [netlib] Error 2
```
I am using clang and gfortran v7 ('make CC=clang FC=gfortran-mp-7'), which has worked fine for versions 0.2.x, including the current release branch (0.2.21).

--- a/cmake/lapacke.cmake
+++ b/cmake/lapacke.cmake
@@ -70,6 +70,8 @@ set(CSRC
   lapacke_cgeqlf_work.c
   lapacke_cgeqp3.c
   lapacke_cgeqp3_work.c
+  lapacke_cgeqpf.c
+  lapacke_cgeqpf_work.c
   lapacke_cgeqr.c
   lapacke_cgeqr_work.c
   lapacke_cgeqr2.c
@@ -140,8 +142,12 @@ set(CSRC
   lapacke_cggqrf_work.c
   lapacke_cggrqf.c
   lapacke_cggrqf_work.c
+  lapacke_cggsvd.c
+  lapacke_cggsvd_work.c
   lapacke_cggsvd3.c
   lapacke_cggsvd3_work.c
+  lapacke_cggsvp.c
+  lapacke_cggsvp_work.c
   lapacke_cggsvp3.c
   lapacke_cggsvp3_work.c
   lapacke_cgtcon.c
@@ -683,6 +689,8 @@ set(DSRC
   lapacke_dgeqlf_work.c
   lapacke_dgeqp3.c
   lapacke_dgeqp3_work.c
+  lapacke_dgeqpf.c
+  lapacke_dgeqpf_work.c
   lapacke_dgeqr.c
   lapacke_dgeqr_work.c
   lapacke_dgeqr2.c
@@ -753,8 +761,12 @@ set(DSRC
   lapacke_dggqrf_work.c
   lapacke_dggrqf.c
   lapacke_dggrqf_work.c
+  lapacke_dggsvd.c
+  lapacke_dggsvd_work.c
   lapacke_dggsvd3.c
   lapacke_dggsvd3_work.c
+  lapacke_dggsvp.c
+  lapacke_dggsvp_work.c
   lapacke_dggsvp3.c
   lapacke_dggsvp3_work.c
   lapacke_dgtcon.c
@@ -1249,6 +1261,8 @@ set(SSRC
   lapacke_sgeqlf_work.c
   lapacke_sgeqp3.c
   lapacke_sgeqp3_work.c
+  lapacke_sgeqpf.c
+  lapacke_sgeqpf_work.c
   lapacke_sgeqr.c
   lapacke_sgeqr_work.c
   lapacke_sgeqr2.c
@@ -1319,8 +1333,12 @@ set(SSRC
   lapacke_sggqrf_work.c
   lapacke_sggrqf.c
   lapacke_sggrqf_work.c
+  lapacke_sggsvd.c
+  lapacke_sggsvd_work.c
   lapacke_sggsvd3.c
   lapacke_sggsvd3_work.c
+  lapacke_sggsvp.c
+  lapacke_sggsvp_work.c
   lapacke_sggsvp3.c
   lapacke_sggsvp3_work.c
   lapacke_sgtcon.c
@@ -1809,6 +1827,8 @@ set(ZSRC
   lapacke_zgeqlf_work.c
   lapacke_zgeqp3.c
   lapacke_zgeqp3_work.c
+  lapacke_zgeqpf.c
+  lapacke_zgeqpf_work.c
   lapacke_zgeqr.c
   lapacke_zgeqr_work.c
   lapacke_zgeqr2.c
@@ -1879,8 +1899,12 @@ set(ZSRC
   lapacke_zggqrf_work.c
   lapacke_zggrqf.c
   lapacke_zggrqf_work.c
+  lapacke_zggsvd.c
+  lapacke_zggsvd_work.c
   lapacke_zggsvd3.c
   lapacke_zggsvd3_work.c
+  lapacke_zggsvp.c
+  lapacke_zggsvp_work.c
   lapacke_zggsvp3.c
   lapacke_zggsvp3_work.c
   lapacke_zgtcon.c

--- a/lapack-netlib/LAPACKE/src/Makefile
+++ b/lapack-netlib/LAPACKE/src/Makefile
@@ -2457,9 +2457,15 @@ all: ../../$(LAPACKELIB)
 ../../$(LAPACKELIB): $(OBJ_A) $(OBJ_B) $(DEPRECATED) $(EXTENDED) $(MATGEN)
 	$(ARCH) $(ARCHFLAGS) $@ $(OBJ_A)
 	$(ARCH) $(ARCHFLAGS) $@ $(OBJ_B)
+ifdef BUILD_DEPRECATED
 	$(ARCH) $(ARCHFLAGS) $@ $(DEPRECATED)
+endif
+ifdef (USEXBLAS)
 	$(ARCH) $(ARCHFLAGS) $@ $(EXTENDED)
+endif
+ifdef LAPACKE_WITH_TMG
 	$(ARCH) $(ARCHFLAGS) $@ $(MATGEN)
+endif
 	$(RANLIB) $@
 
 clean: cleanobj

